### DISCUSSION TITLE
Configuring NATS Monitoring: Cluster-Level Alerts with Prometheus Exporter

### DISCUSSION THREAD

**Discussion by mohamedsaleem18** [nats-io/nats-server#6145] [Q&A]
The NATS cluster consists of three NATS servers (version 2.10.19, Alpine) with JetStream enabled, running on Red Hat OpenShift. While setting up monitoring alerts for the NATS cluster, monitoring is currently available for each individual NATS server via its monitoring endpoint. Could you please guide me on how to set up monitoring alerts at the cluster level (consolidated) in addition to the individual NATS server level alerts using the NATS Prometheus exporter?

**Comment by neilalexander** [comment]
Have you looked at NATS Surveyor? https://github.com/nats-io/nats-surveyor

**Reply by mohamedsaleem18** [reply]
Based on the documentation, it appears that NATS Surveyor runs as an additional component. Since our OpenShift ecosystem uses the Prometheus exporter by default for all applications deployed on OpenShift, we prefer to stick with the Prometheus exporter. Are there any advantages to using NATS Surveyor over the NATS Prometheus exporter? Additionally, can NATS Surveyor be deployed on Red Hat OpenShift alongside the NATS server?
